torch/csrc/jit/operator_upgraders/utils.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+#include <string>
+#include <regex>
+#include <vector>
+#include <torch/csrc/jit/operator_upgraders/version_map.h>
+
+namespace torch {
+namespace jit {
+
+static std::string findUpgrader(UpgraderDB upgraders_for_schema, int current_version) {
+    // we want to find the entry which satisfies following two conditions:
+    //    1. the version entry must be greater than current_version
+    //    2. Among the version entries, we need to see if the current version
+    //       is in the upgrader name range
+    for (const auto& upgrader_entry: upgraders_for_schema) {
+        if (upgrader_entry.first > current_version) {
+            auto upgrader_name = upgrader_entry.second;
+            std::regex delimiter("_");
```

Comment:
Likely it's not allowed internally: https://www.internalfb.com/intern/wiki/RegexEngines/

> Both boost::regex and std::regex are known bad libraries. Of the two, std::regex is significantly worse, but both have caused numerous SEVs because of the way they're implemented. Both of these libraries use a recursive backtracking implementation which has exponential time+memory in the worst case.